{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: dplyr\n",
      "Warning message:\n",
      "“package ‘dplyr’ was built under R version 3.5.3”\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Loading required package: stringr\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>sample</th><th scope=col>EFFECT</th><th scope=col>HGVS_P</th><th scope=col>FEATURE</th><th scope=col>position</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MMRF_1021_1_BM</td><td>missense_variant</td><td>p.Gln61Leu</td><td>transcript</td><td>61</td></tr>\n",
       "\t<tr><td>MMRF_1030_1_BM</td><td>missense_variant</td><td>p.Gly13Cys</td><td>transcript</td><td>13</td></tr>\n",
       "\t<tr><td>MMRF_1030_3_BM</td><td>missense_variant</td><td>p.Gly13Cys</td><td>transcript</td><td>13</td></tr>\n",
       "\t<tr><td>MMRF_1032_1_BM</td><td>missense_variant</td><td>p.Gln61His</td><td>transcript</td><td>61</td></tr>\n",
       "\t<tr><td>MMRF_1038_1_BM</td><td>missense_variant</td><td>p.Gly13Asp</td><td>transcript</td><td>13</td></tr>\n",
       "\t<tr><td>MMRF_1049_4_BM</td><td>missense_variant</td><td>p.Gln61His</td><td>transcript</td><td>61</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       " sample & EFFECT & HGVS\\_P & FEATURE & position\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t MMRF\\_1021\\_1\\_BM & missense\\_variant & p.Gln61Leu & transcript & 61\\\\\n",
       "\t MMRF\\_1030\\_1\\_BM & missense\\_variant & p.Gly13Cys & transcript & 13\\\\\n",
       "\t MMRF\\_1030\\_3\\_BM & missense\\_variant & p.Gly13Cys & transcript & 13\\\\\n",
       "\t MMRF\\_1032\\_1\\_BM & missense\\_variant & p.Gln61His & transcript & 61\\\\\n",
       "\t MMRF\\_1038\\_1\\_BM & missense\\_variant & p.Gly13Asp & transcript & 13\\\\\n",
       "\t MMRF\\_1049\\_4\\_BM & missense\\_variant & p.Gln61His & transcript & 61\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| sample &lt;chr&gt; | EFFECT &lt;chr&gt; | HGVS_P &lt;chr&gt; | FEATURE &lt;chr&gt; | position &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| MMRF_1021_1_BM | missense_variant | p.Gln61Leu | transcript | 61 |\n",
       "| MMRF_1030_1_BM | missense_variant | p.Gly13Cys | transcript | 13 |\n",
       "| MMRF_1030_3_BM | missense_variant | p.Gly13Cys | transcript | 13 |\n",
       "| MMRF_1032_1_BM | missense_variant | p.Gln61His | transcript | 61 |\n",
       "| MMRF_1038_1_BM | missense_variant | p.Gly13Asp | transcript | 13 |\n",
       "| MMRF_1049_4_BM | missense_variant | p.Gln61His | transcript | 61 |\n",
       "\n"
      ],
      "text/plain": [
       "  sample         EFFECT           HGVS_P     FEATURE    position\n",
       "1 MMRF_1021_1_BM missense_variant p.Gln61Leu transcript 61      \n",
       "2 MMRF_1030_1_BM missense_variant p.Gly13Cys transcript 13      \n",
       "3 MMRF_1030_3_BM missense_variant p.Gly13Cys transcript 13      \n",
       "4 MMRF_1032_1_BM missense_variant p.Gln61His transcript 61      \n",
       "5 MMRF_1038_1_BM missense_variant p.Gly13Asp transcript 13      \n",
       "6 MMRF_1049_4_BM missense_variant p.Gln61His transcript 61      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>sample</th><th scope=col>EFFECT</th><th scope=col>HGVS_P</th><th scope=col>FEATURE</th><th scope=col>position</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MMRF_1037_1_BM</td><td>missense_variant</td><td>p.Gln61Lys</td><td>transcript</td><td>61</td></tr>\n",
       "\t<tr><td>MMRF_1054_1_BM</td><td>missense_variant</td><td>p.Gln61Arg</td><td>transcript</td><td>61</td></tr>\n",
       "\t<tr><td>MMRF_1055_1_BM</td><td>missense_variant</td><td>p.Gln61Lys</td><td>transcript</td><td>61</td></tr>\n",
       "\t<tr><td>MMRF_1061_1_BM</td><td>missense_variant</td><td>p.Gln61Arg</td><td>transcript</td><td>61</td></tr>\n",
       "\t<tr><td>MMRF_1061_3_BM</td><td>missense_variant</td><td>p.Gln61Arg</td><td>transcript</td><td>61</td></tr>\n",
       "\t<tr><td>MMRF_1068_1_BM</td><td>missense_variant</td><td>p.Gly13Cys</td><td>transcript</td><td>13</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       " sample & EFFECT & HGVS\\_P & FEATURE & position\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t MMRF\\_1037\\_1\\_BM & missense\\_variant & p.Gln61Lys & transcript & 61\\\\\n",
       "\t MMRF\\_1054\\_1\\_BM & missense\\_variant & p.Gln61Arg & transcript & 61\\\\\n",
       "\t MMRF\\_1055\\_1\\_BM & missense\\_variant & p.Gln61Lys & transcript & 61\\\\\n",
       "\t MMRF\\_1061\\_1\\_BM & missense\\_variant & p.Gln61Arg & transcript & 61\\\\\n",
       "\t MMRF\\_1061\\_3\\_BM & missense\\_variant & p.Gln61Arg & transcript & 61\\\\\n",
       "\t MMRF\\_1068\\_1\\_BM & missense\\_variant & p.Gly13Cys & transcript & 13\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| sample &lt;chr&gt; | EFFECT &lt;chr&gt; | HGVS_P &lt;chr&gt; | FEATURE &lt;chr&gt; | position &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| MMRF_1037_1_BM | missense_variant | p.Gln61Lys | transcript | 61 |\n",
       "| MMRF_1054_1_BM | missense_variant | p.Gln61Arg | transcript | 61 |\n",
       "| MMRF_1055_1_BM | missense_variant | p.Gln61Lys | transcript | 61 |\n",
       "| MMRF_1061_1_BM | missense_variant | p.Gln61Arg | transcript | 61 |\n",
       "| MMRF_1061_3_BM | missense_variant | p.Gln61Arg | transcript | 61 |\n",
       "| MMRF_1068_1_BM | missense_variant | p.Gly13Cys | transcript | 13 |\n",
       "\n"
      ],
      "text/plain": [
       "  sample         EFFECT           HGVS_P     FEATURE    position\n",
       "1 MMRF_1037_1_BM missense_variant p.Gln61Lys transcript 61      \n",
       "2 MMRF_1054_1_BM missense_variant p.Gln61Arg transcript 61      \n",
       "3 MMRF_1055_1_BM missense_variant p.Gln61Lys transcript 61      \n",
       "4 MMRF_1061_1_BM missense_variant p.Gln61Arg transcript 61      \n",
       "5 MMRF_1061_3_BM missense_variant p.Gln61Arg transcript 61      \n",
       "6 MMRF_1068_1_BM missense_variant p.Gly13Cys transcript 13      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "setwd('~/Downloads/proj/')\n",
    "\n",
    "variant = read.delim(\"MMRF_CoMMpass_IA13a_IGV_All_Canonical_Variants.mut\", stringsAsFactors = F)\n",
    "\n",
    "# Select NRAS and KRAS variants\n",
    "require(dplyr)\n",
    "require(stringr)\n",
    "KRAS_mut = filter(variant, GENE == \"KRAS\") %>%\n",
    "  select(sample, EFFECT, HGVS_P, FEATURE) %>%\n",
    "  filter(EFFECT == \"missense_variant\") %>%\n",
    "  mutate(position = str_extract(HGVS_P, \"[0-9]+\")) %>%\n",
    "  filter(position %in% c(12, 13, 61))\n",
    "NRAS_mut = filter(variant, GENE == \"NRAS\") %>%\n",
    "  select(sample, EFFECT, HGVS_P, FEATURE) %>%\n",
    "  filter(EFFECT == \"missense_variant\") %>%\n",
    "  mutate(position = str_extract(HGVS_P, \"[0-9]+\")) %>%\n",
    "  filter(position %in% c(12, 13, 61))\n",
    "\n",
    "\n",
    "head(KRAS_mut)\n",
    "head(NRAS_mut)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample</th><th scope=col>EFFECT</th><th scope=col>HGVS_P</th><th scope=col>FEATURE</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>345</th><td>MMRF_2815_1_BM</td><td>missense_variant</td><td>p.Lys117Asn</td><td>transcript</td></tr>\n",
       "\t<tr><th scope=row>346</th><td>MMRF_2820_1_BM</td><td>missense_variant</td><td>p.Gly13Asp </td><td>transcript</td></tr>\n",
       "\t<tr><th scope=row>347</th><td>MMRF_2821_1_BM</td><td>missense_variant</td><td>p.Gln61Leu </td><td>transcript</td></tr>\n",
       "\t<tr><th scope=row>348</th><td>MMRF_2828_1_BM</td><td>missense_variant</td><td>p.Gly13Asp </td><td>transcript</td></tr>\n",
       "\t<tr><th scope=row>349</th><td>MMRF_2838_1_BM</td><td>missense_variant</td><td>p.Gly12Arg </td><td>transcript</td></tr>\n",
       "\t<tr><th scope=row>350</th><td>MMRF_2840_1_BM</td><td>missense_variant</td><td>p.Gly12Ala </td><td>transcript</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & sample & EFFECT & HGVS\\_P & FEATURE\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t345 & MMRF\\_2815\\_1\\_BM & missense\\_variant & p.Lys117Asn & transcript\\\\\n",
       "\t346 & MMRF\\_2820\\_1\\_BM & missense\\_variant & p.Gly13Asp  & transcript\\\\\n",
       "\t347 & MMRF\\_2821\\_1\\_BM & missense\\_variant & p.Gln61Leu  & transcript\\\\\n",
       "\t348 & MMRF\\_2828\\_1\\_BM & missense\\_variant & p.Gly13Asp  & transcript\\\\\n",
       "\t349 & MMRF\\_2838\\_1\\_BM & missense\\_variant & p.Gly12Arg  & transcript\\\\\n",
       "\t350 & MMRF\\_2840\\_1\\_BM & missense\\_variant & p.Gly12Ala  & transcript\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | sample &lt;chr&gt; | EFFECT &lt;chr&gt; | HGVS_P &lt;chr&gt; | FEATURE &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 345 | MMRF_2815_1_BM | missense_variant | p.Lys117Asn | transcript |\n",
       "| 346 | MMRF_2820_1_BM | missense_variant | p.Gly13Asp  | transcript |\n",
       "| 347 | MMRF_2821_1_BM | missense_variant | p.Gln61Leu  | transcript |\n",
       "| 348 | MMRF_2828_1_BM | missense_variant | p.Gly13Asp  | transcript |\n",
       "| 349 | MMRF_2838_1_BM | missense_variant | p.Gly12Arg  | transcript |\n",
       "| 350 | MMRF_2840_1_BM | missense_variant | p.Gly12Ala  | transcript |\n",
       "\n"
      ],
      "text/plain": [
       "    sample         EFFECT           HGVS_P      FEATURE   \n",
       "345 MMRF_2815_1_BM missense_variant p.Lys117Asn transcript\n",
       "346 MMRF_2820_1_BM missense_variant p.Gly13Asp  transcript\n",
       "347 MMRF_2821_1_BM missense_variant p.Gln61Leu  transcript\n",
       "348 MMRF_2828_1_BM missense_variant p.Gly13Asp  transcript\n",
       "349 MMRF_2838_1_BM missense_variant p.Gly12Arg  transcript\n",
       "350 MMRF_2840_1_BM missense_variant p.Gly12Ala  transcript"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'missense_variant'</li>\n",
       "\t<li>'downstream_gene_variant'</li>\n",
       "\t<li>'synonymous_variant'</li>\n",
       "\t<li>'intragenic_variant'</li>\n",
       "\t<li>'intron_variant'</li>\n",
       "\t<li>'3_prime_UTR_variant'</li>\n",
       "\t<li>'disruptive_inframe_deletion'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'missense\\_variant'\n",
       "\\item 'downstream\\_gene\\_variant'\n",
       "\\item 'synonymous\\_variant'\n",
       "\\item 'intragenic\\_variant'\n",
       "\\item 'intron\\_variant'\n",
       "\\item '3\\_prime\\_UTR\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_deletion'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'missense_variant'\n",
       "2. 'downstream_gene_variant'\n",
       "3. 'synonymous_variant'\n",
       "4. 'intragenic_variant'\n",
       "5. 'intron_variant'\n",
       "6. '3_prime_UTR_variant'\n",
       "7. 'disruptive_inframe_deletion'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"missense_variant\"            \"downstream_gene_variant\"    \n",
       "[3] \"synonymous_variant\"          \"intragenic_variant\"         \n",
       "[5] \"intron_variant\"              \"3_prime_UTR_variant\"        \n",
       "[7] \"disruptive_inframe_deletion\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "KRAS_mut = filter(variant, GENE == \"KRAS\") %>%\n",
    "  select(sample, EFFECT, HGVS_P, FEATURE)# %>%\n",
    "  #filter(EFFECT %in% c(\"missense_variant\",\n",
    "   #                    \"frameshift_variant\",\n",
    "    #                   \"stop_gained\",\n",
    "     #                  \"start_lost\",\n",
    "      #                 \"stop_lost\"))# %>%\n",
    " # mutate(position = str_extract(HGVS_P, \"[0-9]+\")) %>%\n",
    "  #filter(position %in% c(12, 13, 61))\n",
    "tail(KRAS_mut)\n",
    "unique(KRAS_mut$EFFECT)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 72</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>sample</th><th scope=col>EFFECT</th><th scope=col>GENE</th><th scope=col>ID</th><th scope=col>REF</th><th scope=col>ALT</th><th scope=col>X1000G</th><th scope=col>⋯</th><th scope=col>CDS_POS</th><th scope=col>CDS_LEN</th><th scope=col>AA_POS</th><th scope=col>AA_LEN</th><th scope=col>DISTANCE</th><th scope=col>ERRORS</th><th scope=col>LOF_GENE</th><th scope=col>LOF_GENEID</th><th scope=col>LOF_NUMTR</th><th scope=col>LOF_PERC</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>12</td><td>25380274</td><td>25380274</td><td>MMRF_1021_1_BM</td><td>missense_variant</td><td>KRAS</td><td>.          </td><td>C</td><td>T</td><td>.</td><td>⋯</td><td>184</td><td>570</td><td>62</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25380276</td><td>25380276</td><td>MMRF_1021_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs121913240</td><td>T</td><td>A</td><td>.</td><td>⋯</td><td>182</td><td>570</td><td>61</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25398282</td><td>25398282</td><td>MMRF_1030_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs121913535</td><td>C</td><td>A</td><td>.</td><td>⋯</td><td> 37</td><td>570</td><td>13</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25380275</td><td>25380275</td><td>MMRF_1032_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs17851045 </td><td>T</td><td>G</td><td>.</td><td>⋯</td><td>183</td><td>570</td><td>61</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25398281</td><td>25398281</td><td>MMRF_1038_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs112445441</td><td>C</td><td>T</td><td>.</td><td>⋯</td><td> 38</td><td>570</td><td>13</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25398284</td><td>25398284</td><td>MMRF_1078_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs121913529</td><td>C</td><td>T</td><td>.</td><td>⋯</td><td> 35</td><td>570</td><td>12</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 72\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       " chr & start & end & sample & EFFECT & GENE & ID & REF & ALT & X1000G & COSMIC\\_C & COSMIC\\_NC & EXAC & DB & NHLBI & CALLER\\_COUNT & SEURAT & MUTECT & STRELKA & RNA\\_REF\\_COUNT & RNA\\_ALT\\_COUNT & RNA\\_ALT\\_FREQ & NORMAL\\_REF\\_AD & NORMAL\\_ALT\\_AD & NORMAL\\_ALT\\_FREQ & NORMAL\\_DP & TUMOR\\_REF\\_AD & TUMOR\\_ALT\\_AD & TUMOR\\_ALT\\_FREQ & TUMOR\\_DP & dbNSFP\\_GERP\\_RS & dbNSFP\\_LRT\\_score & dbNSFP\\_MutationAssessor\\_score & dbNSFP\\_FATHMM\\_score & dbNSFP\\_GERP\\_NR & dbNSFP\\_CADD\\_raw & dbNSFP\\_MutationTaster\\_score & dbNSFP\\_CADD\\_raw\\_rankscore & dbNSFP\\_CADD\\_phred & dbNSFP\\_MetaSVM\\_score & dbNSFP\\_MetaSVM\\_rankscore & dbNSFP\\_MetaSVM\\_pred & dbNSFP\\_MetaLR\\_score & dbNSFP\\_MetaLR\\_rankscore & dbNSFP\\_MetaLR\\_pred & dbNSFP\\_Reliability\\_index & dbNSFP\\_Interpro\\_domain & dbNSFP\\_Polyphen2\\_HDIV\\_score & dbNSFP\\_SIFT\\_score & dbNSFP\\_Polyphen2\\_HVAR\\_score & dbNSFP\\_Polyphen2\\_HVAR\\_pred & ALLELE & IMPACT & GENEID & FEATURE & FEATUREID & BIOTYPE & RANK & HGVS\\_C & HGVS\\_P & CDNA\\_POS & CDNA\\_LEN & CDS\\_POS & CDS\\_LEN & AA\\_POS & AA\\_LEN & DISTANCE & ERRORS & LOF\\_GENE & LOF\\_GENEID & LOF\\_NUMTR & LOF\\_PERC\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <int> & <int> & <int> & <dbl> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 12 & 25380274 & 25380274 & MMRF\\_1021\\_1\\_BM & missense\\_variant & KRAS & .           & C & T & . & true & . & .    & .    & . & 3 & true & true & true & 51 & 19 & 0.271429 &  50 & 0 & 0 &  50 &  35 & 11 & 0.239 &  46 & 5.77 & 0.000000 & 4.26  & -1.79,-1.79             & 5.77 & 2.699938 & 1.000 & 0.65720 & 22.5 & 0.8008 & 0.94845 & D & 0.7612 & 0.92479 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.904,0.942 & 0.021,0.021             & 0.293,0.562 & B,P & T & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 3 & c.184G>A & p.Glu62Lys & 248 & 1119 & 184 & 570 & 62 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25380276 & 25380276 & MMRF\\_1021\\_1\\_BM & missense\\_variant & KRAS & rs121913240 & T & A & . & true & . & .    & true & . & 3 & true & true & true & 53 & 19 & 0.263889 &  50 & 0 & 0 &  50 &  36 & 12 & 0.250 &  48 & 5.77 & 0.000000 & 3.805 & -1.79,-1.79             & 5.77 & 2.685591 & 1.000 & 0.65476 & 22.5 & 0.7532 & 0.94295 & D & 0.7495 & 0.92097 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.984,0.812 & 0.002,0.002             & 0.732,0.69  & P,P & A & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 3 & c.182A>T & p.Gln61Leu & 246 & 1119 & 182 & 570 & 61 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25398282 & 25398282 & MMRF\\_1030\\_1\\_BM & missense\\_variant & KRAS & rs121913535 & C & A & . & true & . & .    & true & . & 3 & true & true & true & 20 & 14 & 0.411765 & 214 & 0 & 0 & 214 &  88 & 66 & 0.429 & 154 & 5.68 & 0.000000 & 2.97  & -1.76,-0.81,-0.81,-0.81 & 5.68 & 6.126117 & 1.000 & 0.96998 & 34   & 0.6946 & 0.93619 & D & 0.7692 & 0.92740 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 1.0,1.0     & 0.0,0.0,0.0,0.0         & 0.981,0.997 & D,D & A & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 2 & c.37G>T  & p.Gly13Cys & 101 & 1119 &  37 & 570 & 13 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25380275 & 25380275 & MMRF\\_1032\\_1\\_BM & missense\\_variant & KRAS & rs17851045  & T & G & . & true & . & .    & true & . & 3 & true & true & true & 42 & 8  & 0.16     & 171 & 0 & 0 & 171 & 104 & 45 & 0.302 & 149 & 5.77 & 0.000000 & 3.18  & -1.81,-1.81             & 5.77 & 2.435859 & 1.000 & 0.59457 & 22.1 & 0.2643 & 0.87829 & D & 0.6134 & 0.87265 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.411,0.09  & 0.007,0.008             & 0.092,0.151 & B,B & G & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 3 & c.183A>C & p.Gln61His & 247 & 1119 & 183 & 570 & 61 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25398281 & 25398281 & MMRF\\_1038\\_1\\_BM & missense\\_variant & KRAS & rs112445441 & C & T & . & true & . & .    & true & . & 3 & true & true & true & 14 & 16 & 0.533333 & 132 & 0 & 0 & 132 &  48 & 34 & 0.415 &  82 & 5.68 & 0.000000 & 3.375 & -1.65,-0.7,-0.7,-0.7    & 5.68 & 5.462911 & 1.000 & 0.92850 & 29.4 & 0.5692 & 0.92106 & D & 0.6901 & 0.90079 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.215,0.803 & 0.029,0.012,0.032,0.016 & 0.175,0.506 & B,P & T & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 2 & c.38G>A  & p.Gly13Asp & 102 & 1119 &  38 & 570 & 13 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25398284 & 25398284 & MMRF\\_1078\\_1\\_BM & missense\\_variant & KRAS & rs121913529 & C & T & . & true & . & true & true & . & 3 & true & true & true & 6  & 1  & 0.142857 & 138 & 0 & 0 & 138 &  75 &  7 & 0.085 &  82 & 5.68 & 0.000000 & 3.245 & -1.19,-1.19,-1.19,-1.19 & 5.68 & 4.690595 & 1.000 & 0.83872 & 25.8 & 0.2670 & 0.87870 & D & 0.5809 & 0.85919 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.385,0.517 & 0.01,0.018,0.01,0.024   & 0.257,0.361 & B,B & T & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 2 & c.35G>A  & p.Gly12Asp &  99 & 1119 &  35 & 570 & 12 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 72\n",
       "\n",
       "| chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | sample &lt;chr&gt; | EFFECT &lt;chr&gt; | GENE &lt;chr&gt; | ID &lt;chr&gt; | REF &lt;chr&gt; | ALT &lt;chr&gt; | X1000G &lt;chr&gt; | ⋯ ⋯ | CDS_POS &lt;int&gt; | CDS_LEN &lt;int&gt; | AA_POS &lt;int&gt; | AA_LEN &lt;int&gt; | DISTANCE &lt;int&gt; | ERRORS &lt;chr&gt; | LOF_GENE &lt;chr&gt; | LOF_GENEID &lt;chr&gt; | LOF_NUMTR &lt;chr&gt; | LOF_PERC &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 12 | 25380274 | 25380274 | MMRF_1021_1_BM | missense_variant | KRAS | .           | C | T | . | ⋯ | 184 | 570 | 62 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25380276 | 25380276 | MMRF_1021_1_BM | missense_variant | KRAS | rs121913240 | T | A | . | ⋯ | 182 | 570 | 61 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25398282 | 25398282 | MMRF_1030_1_BM | missense_variant | KRAS | rs121913535 | C | A | . | ⋯ |  37 | 570 | 13 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25380275 | 25380275 | MMRF_1032_1_BM | missense_variant | KRAS | rs17851045  | T | G | . | ⋯ | 183 | 570 | 61 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25398281 | 25398281 | MMRF_1038_1_BM | missense_variant | KRAS | rs112445441 | C | T | . | ⋯ |  38 | 570 | 13 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25398284 | 25398284 | MMRF_1078_1_BM | missense_variant | KRAS | rs121913529 | C | T | . | ⋯ |  35 | 570 | 12 | 189 | 0 | . | . | . | . | . |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start    end      sample         EFFECT           GENE ID          REF\n",
       "1 12  25380274 25380274 MMRF_1021_1_BM missense_variant KRAS .           C  \n",
       "2 12  25380276 25380276 MMRF_1021_1_BM missense_variant KRAS rs121913240 T  \n",
       "3 12  25398282 25398282 MMRF_1030_1_BM missense_variant KRAS rs121913535 C  \n",
       "4 12  25380275 25380275 MMRF_1032_1_BM missense_variant KRAS rs17851045  T  \n",
       "5 12  25398281 25398281 MMRF_1038_1_BM missense_variant KRAS rs112445441 C  \n",
       "6 12  25398284 25398284 MMRF_1078_1_BM missense_variant KRAS rs121913529 C  \n",
       "  ALT X1000G ⋯ CDS_POS CDS_LEN AA_POS AA_LEN DISTANCE ERRORS LOF_GENE\n",
       "1 T   .      ⋯ 184     570     62     189    0        .      .       \n",
       "2 A   .      ⋯ 182     570     61     189    0        .      .       \n",
       "3 A   .      ⋯  37     570     13     189    0        .      .       \n",
       "4 G   .      ⋯ 183     570     61     189    0        .      .       \n",
       "5 T   .      ⋯  38     570     13     189    0        .      .       \n",
       "6 T   .      ⋯  35     570     12     189    0        .      .       \n",
       "  LOF_GENEID LOF_NUMTR LOF_PERC\n",
       "1 .          .         .       \n",
       "2 .          .         .       \n",
       "3 .          .         .       \n",
       "4 .          .         .       \n",
       "5 .          .         .       \n",
       "6 .          .         .       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "237"
      ],
      "text/latex": [
       "237"
      ],
      "text/markdown": [
       "237"
      ],
      "text/plain": [
       "[1] 237"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "variant2 = filter(variant, EFFECT %in% c(\"missense_variant\",\n",
    "                                         \"frameshift_variant\",\n",
    "                                         \"stop_gained\",\n",
    "                                         \"start_lost\",\n",
    "                                         \"stop_lost\",\n",
    "                                         \"missense_variant&splice_region_variant\",\n",
    "                                         \"frameshift_variant&splice_region_variant\",\n",
    "                                         \"disruptive_inframe_deletion\",\n",
    "                                         \"stop_gained&splice_region_variant\",\n",
    "                                         \"frameshift_variant&stop_gained\",\n",
    "                                         \"frameshift_variant&splice_acceptor_variant&splice_region_variant&intron_variant\",\n",
    "                                         \"frameshift_variant&splice_donor_variant&splice_region_variant&intron_variant\",\n",
    "                                         \"splice_donor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\",\n",
    "                                         \"stop_lost&splice_region_variant\",\n",
    "                                         \"disruptive_inframe_insertion\",\n",
    "                                         \"stop_gained&disruptive_inframe_deletion\",\n",
    "                                         \"frameshift_variant&start_lost\",\n",
    "                                         \"stop_gained&disruptive_inframe_insertion\",\n",
    "                                         \"disruptive_inframe_deletion&splice_region_variant\",\n",
    "                                         \"disruptive_inframe_insertion&splice_region_variant\",\n",
    "                                         \"start_lost&splice_region_variant\",\n",
    "                                         \"start_lost&inframe_deletion\",\n",
    "                                         \"frameshift_variant&stop_lost\",\n",
    "                                         \"splice_acceptor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\",\n",
    "                                         \"stop_lost&inframe_deletion\")) %>%\n",
    "  filter(str_extract(sample, \"_[0-9]_\") == \"_1_\")  %>%\n",
    "  filter(GENE == 'KRAS')# Limit variants to newly diagnosed patients\n",
    "head(variant2)\n",
    "length(unique(variant2$sample))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 72</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>sample</th><th scope=col>EFFECT</th><th scope=col>GENE</th><th scope=col>ID</th><th scope=col>REF</th><th scope=col>ALT</th><th scope=col>X1000G</th><th scope=col>⋯</th><th scope=col>CDS_POS</th><th scope=col>CDS_LEN</th><th scope=col>AA_POS</th><th scope=col>AA_LEN</th><th scope=col>DISTANCE</th><th scope=col>ERRORS</th><th scope=col>LOF_GENE</th><th scope=col>LOF_GENEID</th><th scope=col>LOF_NUMTR</th><th scope=col>LOF_PERC</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>  3348577</td><td>  3348577</td><td>MMRF_1016_1_BM</td><td>missense_variant                      </td><td>PRDM16 </td><td>rs755588843</td><td>C</td><td>T</td><td>.</td><td>⋯</td><td>3569</td><td>3831</td><td>1190</td><td>1276</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>1</td><td> 94466482</td><td> 94466482</td><td>MMRF_1016_1_BM</td><td>missense_variant&amp;splice_region_variant</td><td><span style=white-space:pre-wrap>ABCA4  </span></td><td><span style=white-space:pre-wrap>.          </span></td><td>A</td><td>G</td><td>.</td><td>⋯</td><td>6389</td><td>6822</td><td>2130</td><td>2273</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>1</td><td>152327653</td><td>152327653</td><td>MMRF_1016_1_BM</td><td>missense_variant                      </td><td>FLG2   </td><td>.          </td><td>C</td><td>G</td><td>.</td><td>⋯</td><td>2609</td><td>7176</td><td> 870</td><td>2391</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>2</td><td> 21364769</td><td> 21364769</td><td>MMRF_1016_1_BM</td><td>missense_variant                      </td><td>TDRD15 </td><td>.          </td><td>A</td><td>T</td><td>.</td><td>⋯</td><td>4430</td><td>5805</td><td>1477</td><td>1934</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>2</td><td> 43452290</td><td> 43452290</td><td>MMRF_1016_1_BM</td><td>missense_variant                      </td><td>ZFP36L2</td><td>.          </td><td>G</td><td>T</td><td>.</td><td>⋯</td><td> 653</td><td>1485</td><td> 218</td><td> 494</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>2</td><td> 43452291</td><td> 43452291</td><td>MMRF_1016_1_BM</td><td>missense_variant                      </td><td>ZFP36L2</td><td>.          </td><td>C</td><td>A</td><td>.</td><td>⋯</td><td> 652</td><td>1485</td><td> 218</td><td> 494</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 72\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       " chr & start & end & sample & EFFECT & GENE & ID & REF & ALT & X1000G & COSMIC\\_C & COSMIC\\_NC & EXAC & DB & NHLBI & CALLER\\_COUNT & SEURAT & MUTECT & STRELKA & RNA\\_REF\\_COUNT & RNA\\_ALT\\_COUNT & RNA\\_ALT\\_FREQ & NORMAL\\_REF\\_AD & NORMAL\\_ALT\\_AD & NORMAL\\_ALT\\_FREQ & NORMAL\\_DP & TUMOR\\_REF\\_AD & TUMOR\\_ALT\\_AD & TUMOR\\_ALT\\_FREQ & TUMOR\\_DP & dbNSFP\\_GERP\\_RS & dbNSFP\\_LRT\\_score & dbNSFP\\_MutationAssessor\\_score & dbNSFP\\_FATHMM\\_score & dbNSFP\\_GERP\\_NR & dbNSFP\\_CADD\\_raw & dbNSFP\\_MutationTaster\\_score & dbNSFP\\_CADD\\_raw\\_rankscore & dbNSFP\\_CADD\\_phred & dbNSFP\\_MetaSVM\\_score & dbNSFP\\_MetaSVM\\_rankscore & dbNSFP\\_MetaSVM\\_pred & dbNSFP\\_MetaLR\\_score & dbNSFP\\_MetaLR\\_rankscore & dbNSFP\\_MetaLR\\_pred & dbNSFP\\_Reliability\\_index & dbNSFP\\_Interpro\\_domain & dbNSFP\\_Polyphen2\\_HDIV\\_score & dbNSFP\\_SIFT\\_score & dbNSFP\\_Polyphen2\\_HVAR\\_score & dbNSFP\\_Polyphen2\\_HVAR\\_pred & ALLELE & IMPACT & GENEID & FEATURE & FEATUREID & BIOTYPE & RANK & HGVS\\_C & HGVS\\_P & CDNA\\_POS & CDNA\\_LEN & CDS\\_POS & CDS\\_LEN & AA\\_POS & AA\\_LEN & DISTANCE & ERRORS & LOF\\_GENE & LOF\\_GENEID & LOF\\_NUMTR & LOF\\_PERC\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <int> & <int> & <int> & <dbl> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 &   3348577 &   3348577 & MMRF\\_1016\\_1\\_BM & missense\\_variant                       & PRDM16  & rs755588843 & C & T & . & . & . & true & true & . & 2 & .    & true & true & . & . & . &  46 & 0 & 0 &  46 &  29 &  5 & 0.147 &  34 & 3.23 & 0.000136 & 2.08  & 3.41,3.42,3.26,3.26,3.42,3.38,3.38 & 5.11 & 5.34666  & 1 & 0.91457 & 28.6  & -1.1081 & 0.03025 & T & 0.0325 & 0.13415 & T & 10 & .                                                    & 0.021,0.986,0.884,0.977 & 0.035,0.035,0.057,0.057,0.035,0.042,0.04,0.058 & 0.004,0.542,0.231,0.341 & B,P,B,B & T & MODERATE & ENSG00000142611 & transcript & ENST00000270722 & protein\\_coding & 16 & c.3569C>T & p.Pro1190Leu & 3618 & 8690 & 3569 & 3831 & 1190 & 1276 & 0 & . & . & . & . & .\\\\\n",
       "\t 1 &  94466482 &  94466482 & MMRF\\_1016\\_1\\_BM & missense\\_variant\\&splice\\_region\\_variant & ABCA4   & .           & A & G & . & . & . & .    & .    & . & 3 & true & true & true & . & . & . &  86 & 0 & 0 &  86 &  23 & 26 & 0.531 &  49 & 5.84 & 0.540752 & 2.905 & -5.72,-5.72,-5.72                  & 5.84 & 5.16588  & 1 & 0.89186 & 27.6  & 1.1299  & 0.99928 & D & 0.9694 & 0.99111 & D & 10 & ATPase\\_\\_AAA+\\_type\\_\\_core\\_\\_1\\_,ABC\\_transporter-like\\_\\_1\\_ & 0.949                   & 0.001,0,0.001,0                                & 0.632                   & P       & G & MODERATE & ENSG00000198691 & transcript & ENST00000370225 & protein\\_coding & 47 & c.6389T>C & p.Met2130Thr & 6476 & 7309 & 6389 & 6822 & 2130 & 2273 & 0 & . & . & . & . & .\\\\\n",
       "\t 1 & 152327653 & 152327653 & MMRF\\_1016\\_1\\_BM & missense\\_variant                       & FLG2    & .           & C & G & . & . & . & .    & .    & . & 3 & true & true & true & . & . & . & 214 & 0 & 0 & 214 & 270 & 29 & 0.097 & 299 & 2.52 & .        & 1.645 & 3.66                               & 3.46 & 0.226273 & 1 & 0.12597 & 6.619 & -0.9181 & 0.46582 & T & 0.0052 & 0.01627 & T & 9  & .                                                    & 0.649                   & 0.476                                          & 0.189                   & B       & G & MODERATE & ENSG00000143520 & transcript & ENST00000388718 & protein\\_coding &  3 & c.2609G>C & p.Gly870Ala  & 2682 & 9124 & 2609 & 7176 &  870 & 2391 & 0 & . & . & . & . & .\\\\\n",
       "\t 2 &  21364769 &  21364769 & MMRF\\_1016\\_1\\_BM & missense\\_variant                       & TDRD15  & .           & A & T & . & . & . & .    & .    & . & 3 & true & true & true & . & . & . & 147 & 0 & 0 & 147 & 103 & 22 & 0.176 & 125 & 2.96 & .        & .     & 2.11                               & 5.32 & 1.43284  & . & 0.3144  & 15.63 & -0.9869 & 0.3342  & T & 0.0689 & 0.27967 & T & 5  & .                                                    & .                       & 0.021                                          & .                       & .       & T & MODERATE & ENSG00000218819 & transcript & ENST00000405799 & protein\\_coding &  4 & c.4430A>T & p.Lys1477Met & 4760 & 6135 & 4430 & 5805 & 1477 & 1934 & 0 & . & . & . & . & .\\\\\n",
       "\t 2 &  43452290 &  43452290 & MMRF\\_1016\\_1\\_BM & missense\\_variant                       & ZFP36L2 & .           & G & T & . & . & . & .    & .    & . & 3 & true & true & true & . & . & . & 148 & 0 & 0 & 148 & 102 & 15 & 0.128 & 117 & 3.53 & 0.000246 & 0.28  & 0.88                               & 4.41 & 1.74502  & 1 & 0.38416 & 17.27 & -0.9632 & 0.39057 & T & 0.1279 & 0.44138 & T & 10 & Zinc\\_finger\\_\\_CCCH-type\\_\\_2\\_                           & 1                       & 0.05                                           & 0.994                   & D       & T & MODERATE & ENSG00000152518 & transcript & ENST00000282388 & protein\\_coding &  2 & c.653C>A  & p.Ala218Glu  &  947 & 3696 &  653 & 1485 &  218 &  494 & 0 & . & . & . & . & .\\\\\n",
       "\t 2 &  43452291 &  43452291 & MMRF\\_1016\\_1\\_BM & missense\\_variant                       & ZFP36L2 & .           & C & A & . & . & . & .    & .    & . & 3 & true & true & true & . & . & . & 149 & 0 & 0 & 149 & 106 & 16 & 0.131 & 122 & 3.53 & 0.000246 & 0.925 & 0.86                               & 4.41 & 2.01941  & 1 & 0.46115 & 19.24 & -0.8175 & 0.55218 & T & 0.1828 & 0.53948 & T & 10 & Zinc\\_finger\\_\\_CCCH-type\\_\\_2\\_                           & 0.999                   & 0.047                                          & 0.986                   & D       & A & MODERATE & ENSG00000152518 & transcript & ENST00000282388 & protein\\_coding &  2 & c.652G>T  & p.Ala218Ser  &  946 & 3696 &  652 & 1485 &  218 &  494 & 0 & . & . & . & . & .\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 72\n",
       "\n",
       "| chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | sample &lt;chr&gt; | EFFECT &lt;chr&gt; | GENE &lt;chr&gt; | ID &lt;chr&gt; | REF &lt;chr&gt; | ALT &lt;chr&gt; | X1000G &lt;chr&gt; | ⋯ ⋯ | CDS_POS &lt;int&gt; | CDS_LEN &lt;int&gt; | AA_POS &lt;int&gt; | AA_LEN &lt;int&gt; | DISTANCE &lt;int&gt; | ERRORS &lt;chr&gt; | LOF_GENE &lt;chr&gt; | LOF_GENEID &lt;chr&gt; | LOF_NUMTR &lt;chr&gt; | LOF_PERC &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 |   3348577 |   3348577 | MMRF_1016_1_BM | missense_variant                       | PRDM16  | rs755588843 | C | T | . | ⋯ | 3569 | 3831 | 1190 | 1276 | 0 | . | . | . | . | . |\n",
       "| 1 |  94466482 |  94466482 | MMRF_1016_1_BM | missense_variant&amp;splice_region_variant | ABCA4   | .           | A | G | . | ⋯ | 6389 | 6822 | 2130 | 2273 | 0 | . | . | . | . | . |\n",
       "| 1 | 152327653 | 152327653 | MMRF_1016_1_BM | missense_variant                       | FLG2    | .           | C | G | . | ⋯ | 2609 | 7176 |  870 | 2391 | 0 | . | . | . | . | . |\n",
       "| 2 |  21364769 |  21364769 | MMRF_1016_1_BM | missense_variant                       | TDRD15  | .           | A | T | . | ⋯ | 4430 | 5805 | 1477 | 1934 | 0 | . | . | . | . | . |\n",
       "| 2 |  43452290 |  43452290 | MMRF_1016_1_BM | missense_variant                       | ZFP36L2 | .           | G | T | . | ⋯ |  653 | 1485 |  218 |  494 | 0 | . | . | . | . | . |\n",
       "| 2 |  43452291 |  43452291 | MMRF_1016_1_BM | missense_variant                       | ZFP36L2 | .           | C | A | . | ⋯ |  652 | 1485 |  218 |  494 | 0 | . | . | . | . | . |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start     end       sample         EFFECT                                \n",
       "1 1     3348577   3348577 MMRF_1016_1_BM missense_variant                      \n",
       "2 1    94466482  94466482 MMRF_1016_1_BM missense_variant&splice_region_variant\n",
       "3 1   152327653 152327653 MMRF_1016_1_BM missense_variant                      \n",
       "4 2    21364769  21364769 MMRF_1016_1_BM missense_variant                      \n",
       "5 2    43452290  43452290 MMRF_1016_1_BM missense_variant                      \n",
       "6 2    43452291  43452291 MMRF_1016_1_BM missense_variant                      \n",
       "  GENE    ID          REF ALT X1000G ⋯ CDS_POS CDS_LEN AA_POS AA_LEN DISTANCE\n",
       "1 PRDM16  rs755588843 C   T   .      ⋯ 3569    3831    1190   1276   0       \n",
       "2 ABCA4   .           A   G   .      ⋯ 6389    6822    2130   2273   0       \n",
       "3 FLG2    .           C   G   .      ⋯ 2609    7176     870   2391   0       \n",
       "4 TDRD15  .           A   T   .      ⋯ 4430    5805    1477   1934   0       \n",
       "5 ZFP36L2 .           G   T   .      ⋯  653    1485     218    494   0       \n",
       "6 ZFP36L2 .           C   A   .      ⋯  652    1485     218    494   0       \n",
       "  ERRORS LOF_GENE LOF_GENEID LOF_NUMTR LOF_PERC\n",
       "1 .      .        .          .         .       \n",
       "2 .      .        .          .         .       \n",
       "3 .      .        .          .         .       \n",
       "4 .      .        .          .         .       \n",
       "5 .      .        .          .         .       \n",
       "6 .      .        .          .         .       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "IGHV2-70  IGLV3-1 IGHV1-69    IGHJ6    IGHJ5  IGKV1-5  IGKV4-1    IGHJ4 \n",
       "    2670     1700      407      402      387      353      309      305 \n",
       "    KRAS IGLV3-25 IGHV3-33     NRAS IGHV3-30 IGHV3-23    IGLL5 IGLV4-60 \n",
       "     266      234      218      218      214      182      182      180 \n",
       " IGHV3-9    IGKJ5    IGKJ3    IGKJ4      TTN IGLV3-21    IGHJ3    IGKJ2 \n",
       "     170      157      156      148      145      132      126      120 \n",
       "IGLV5-45 IGLV1-44  IGHV2-5     DIS3   FAM46C  IGHV3-7 IGHV5-51 IGLV2-14 \n",
       "     117      112      103      101      100       97       97       97 \n",
       "   MUC16 IGLV6-57    TRAF3 IGLV1-47 IGLV4-69 IGLV2-11 IGLV3-10    IGKJ1 \n",
       "      96       87       87       82       81       77       73       72 \n",
       "IGHV4-34     BRAF IGKV3-20 IGHV3-74 IGLV2-23 IGHV4-39 IGHV4-31 IGLV1-40 \n",
       "      71       68       68       67       67       66       64       64 \n",
       "IGLV3-19  IGKV1-9     FAT4     PCLO IGHV3-21 IGLV1-51     RYR2    CSMD3 \n",
       "      64       62       60       60       59       57       57       56 \n",
       "IGLV7-46    LRP1B     RYR1     FAT3  IGKV5-2  IGLV4-3 IGHV3-43 IGHV1-24 \n",
       "      56       56       56       54       54       54       53       51 \n",
       "    FAT1 IGHV1-46  IGHV1-8    FSIP2 IGKV2-30      FLG IGHV1-18  IGHV1-2 \n",
       "      50       50       50       49       48       47       47       47 \n",
       "   DNAH5     TP53    DUSP2 HIST1H1E    USH2A     EGR1  IGLV3-9  IGLV2-8 \n",
       "      46       46       45       43       43       42       42       41 \n",
       "   HMCN1 IGHV3-49    ZFHX4    GPR98      ATM     RYR3  IGKV1-6 IGLV3-16 \n",
       "      39       39       39       38       37       37       36       36 \n",
       "   CSMD1    IGHJ2    DNAH9    FGFR3  IGHV1-3 IGKV1D-8    KALRN      MAX \n",
       "      34       34       33       33       33       33       33       33 \n",
       "   MUC17    SYNE1      DST    HUWE1 \n",
       "      33       33       32       32 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Identify common variants in newly diagnosed patients (top 100)\n",
    "variant2 = filter(variant, EFFECT %in% c(\"missense_variant\",\n",
    "                                         \"frameshift_variant\",\n",
    "                                         \"stop_gained\",\n",
    "                                         \"start_lost\",\n",
    "                                         \"stop_lost\",\n",
    "                                         \"missense_variant&splice_region_variant\",\n",
    "                                         \"frameshift_variant&splice_region_variant\",\n",
    "                                         \"disruptive_inframe_deletion\",\n",
    "                                         \"stop_gained&splice_region_variant\",\n",
    "                                         \"frameshift_variant&stop_gained\",\n",
    "                                         \"frameshift_variant&splice_acceptor_variant&splice_region_variant&intron_variant\",\n",
    "                                         \"frameshift_variant&splice_donor_variant&splice_region_variant&intron_variant\",\n",
    "                                         \"splice_donor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\",\n",
    "                                         \"stop_lost&splice_region_variant\",\n",
    "                                         \"disruptive_inframe_insertion\",\n",
    "                                         \"stop_gained&disruptive_inframe_deletion\",\n",
    "                                         \"frameshift_variant&start_lost\",\n",
    "                                         \"stop_gained&disruptive_inframe_insertion\",\n",
    "                                         \"disruptive_inframe_deletion&splice_region_variant\",\n",
    "                                         \"disruptive_inframe_insertion&splice_region_variant\",\n",
    "                                         \"start_lost&splice_region_variant\",\n",
    "                                         \"start_lost&inframe_deletion\",\n",
    "                                         \"frameshift_variant&stop_lost\",\n",
    "                                         \"splice_acceptor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\",\n",
    "                                         \"stop_lost&inframe_deletion\")) %>%\n",
    "  filter(str_extract(sample, \"_[0-9]_\") == \"_1_\")   # Limit variants to newly diagnosed patients\n",
    "\n",
    "variant3= mutate(variant, EFFECT %in% c(\"missense\",\n",
    "                                         \"frameshift\",\n",
    "                                         \"stop\",\n",
    "                                         \"start\")) %>%\n",
    "  filter(str_extract(sample, \"_[0-9]_\") == \"_1_\")   # Limit variants to newly diagnosed patients\n",
    "\n",
    "CV = table(variant2$GENE)\n",
    "CV = sort(CV, decreasing = T)[1:100]\n",
    "\n",
    "head(variant2)\n",
    "CV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 72</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>sample</th><th scope=col>EFFECT</th><th scope=col>GENE</th><th scope=col>ID</th><th scope=col>REF</th><th scope=col>ALT</th><th scope=col>X1000G</th><th scope=col>⋯</th><th scope=col>CDS_POS</th><th scope=col>CDS_LEN</th><th scope=col>AA_POS</th><th scope=col>AA_LEN</th><th scope=col>DISTANCE</th><th scope=col>ERRORS</th><th scope=col>LOF_GENE</th><th scope=col>LOF_GENEID</th><th scope=col>LOF_NUMTR</th><th scope=col>LOF_PERC</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>12</td><td>25380274</td><td>25380274</td><td>MMRF_1021_1_BM</td><td>missense_variant</td><td>KRAS</td><td>.          </td><td>C</td><td>T</td><td>.</td><td>⋯</td><td>184</td><td>570</td><td>62</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25380276</td><td>25380276</td><td>MMRF_1021_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs121913240</td><td>T</td><td>A</td><td>.</td><td>⋯</td><td>182</td><td>570</td><td>61</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25398282</td><td>25398282</td><td>MMRF_1030_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs121913535</td><td>C</td><td>A</td><td>.</td><td>⋯</td><td> 37</td><td>570</td><td>13</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25380275</td><td>25380275</td><td>MMRF_1032_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs17851045 </td><td>T</td><td>G</td><td>.</td><td>⋯</td><td>183</td><td>570</td><td>61</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25398281</td><td>25398281</td><td>MMRF_1038_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs112445441</td><td>C</td><td>T</td><td>.</td><td>⋯</td><td> 38</td><td>570</td><td>13</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "\t<tr><td>12</td><td>25398284</td><td>25398284</td><td>MMRF_1078_1_BM</td><td>missense_variant</td><td>KRAS</td><td>rs121913529</td><td>C</td><td>T</td><td>.</td><td>⋯</td><td> 35</td><td>570</td><td>12</td><td>189</td><td>0</td><td>.</td><td>.</td><td>.</td><td>.</td><td>.</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 72\n",
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       " chr & start & end & sample & EFFECT & GENE & ID & REF & ALT & X1000G & COSMIC\\_C & COSMIC\\_NC & EXAC & DB & NHLBI & CALLER\\_COUNT & SEURAT & MUTECT & STRELKA & RNA\\_REF\\_COUNT & RNA\\_ALT\\_COUNT & RNA\\_ALT\\_FREQ & NORMAL\\_REF\\_AD & NORMAL\\_ALT\\_AD & NORMAL\\_ALT\\_FREQ & NORMAL\\_DP & TUMOR\\_REF\\_AD & TUMOR\\_ALT\\_AD & TUMOR\\_ALT\\_FREQ & TUMOR\\_DP & dbNSFP\\_GERP\\_RS & dbNSFP\\_LRT\\_score & dbNSFP\\_MutationAssessor\\_score & dbNSFP\\_FATHMM\\_score & dbNSFP\\_GERP\\_NR & dbNSFP\\_CADD\\_raw & dbNSFP\\_MutationTaster\\_score & dbNSFP\\_CADD\\_raw\\_rankscore & dbNSFP\\_CADD\\_phred & dbNSFP\\_MetaSVM\\_score & dbNSFP\\_MetaSVM\\_rankscore & dbNSFP\\_MetaSVM\\_pred & dbNSFP\\_MetaLR\\_score & dbNSFP\\_MetaLR\\_rankscore & dbNSFP\\_MetaLR\\_pred & dbNSFP\\_Reliability\\_index & dbNSFP\\_Interpro\\_domain & dbNSFP\\_Polyphen2\\_HDIV\\_score & dbNSFP\\_SIFT\\_score & dbNSFP\\_Polyphen2\\_HVAR\\_score & dbNSFP\\_Polyphen2\\_HVAR\\_pred & ALLELE & IMPACT & GENEID & FEATURE & FEATUREID & BIOTYPE & RANK & HGVS\\_C & HGVS\\_P & CDNA\\_POS & CDNA\\_LEN & CDS\\_POS & CDS\\_LEN & AA\\_POS & AA\\_LEN & DISTANCE & ERRORS & LOF\\_GENE & LOF\\_GENEID & LOF\\_NUMTR & LOF\\_PERC\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <int> & <int> & <int> & <dbl> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 12 & 25380274 & 25380274 & MMRF\\_1021\\_1\\_BM & missense\\_variant & KRAS & .           & C & T & . & true & . & .    & .    & . & 3 & true & true & true & 51 & 19 & 0.271429 &  50 & 0 & 0 &  50 &  35 & 11 & 0.239 &  46 & 5.77 & 0.000000 & 4.26  & -1.79,-1.79             & 5.77 & 2.699938 & 1.000 & 0.65720 & 22.5 & 0.8008 & 0.94845 & D & 0.7612 & 0.92479 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.904,0.942 & 0.021,0.021             & 0.293,0.562 & B,P & T & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 3 & c.184G>A & p.Glu62Lys & 248 & 1119 & 184 & 570 & 62 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25380276 & 25380276 & MMRF\\_1021\\_1\\_BM & missense\\_variant & KRAS & rs121913240 & T & A & . & true & . & .    & true & . & 3 & true & true & true & 53 & 19 & 0.263889 &  50 & 0 & 0 &  50 &  36 & 12 & 0.250 &  48 & 5.77 & 0.000000 & 3.805 & -1.79,-1.79             & 5.77 & 2.685591 & 1.000 & 0.65476 & 22.5 & 0.7532 & 0.94295 & D & 0.7495 & 0.92097 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.984,0.812 & 0.002,0.002             & 0.732,0.69  & P,P & A & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 3 & c.182A>T & p.Gln61Leu & 246 & 1119 & 182 & 570 & 61 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25398282 & 25398282 & MMRF\\_1030\\_1\\_BM & missense\\_variant & KRAS & rs121913535 & C & A & . & true & . & .    & true & . & 3 & true & true & true & 20 & 14 & 0.411765 & 214 & 0 & 0 & 214 &  88 & 66 & 0.429 & 154 & 5.68 & 0.000000 & 2.97  & -1.76,-0.81,-0.81,-0.81 & 5.68 & 6.126117 & 1.000 & 0.96998 & 34   & 0.6946 & 0.93619 & D & 0.7692 & 0.92740 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 1.0,1.0     & 0.0,0.0,0.0,0.0         & 0.981,0.997 & D,D & A & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 2 & c.37G>T  & p.Gly13Cys & 101 & 1119 &  37 & 570 & 13 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25380275 & 25380275 & MMRF\\_1032\\_1\\_BM & missense\\_variant & KRAS & rs17851045  & T & G & . & true & . & .    & true & . & 3 & true & true & true & 42 & 8  & 0.16     & 171 & 0 & 0 & 171 & 104 & 45 & 0.302 & 149 & 5.77 & 0.000000 & 3.18  & -1.81,-1.81             & 5.77 & 2.435859 & 1.000 & 0.59457 & 22.1 & 0.2643 & 0.87829 & D & 0.6134 & 0.87265 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.411,0.09  & 0.007,0.008             & 0.092,0.151 & B,B & G & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 3 & c.183A>C & p.Gln61His & 247 & 1119 & 183 & 570 & 61 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25398281 & 25398281 & MMRF\\_1038\\_1\\_BM & missense\\_variant & KRAS & rs112445441 & C & T & . & true & . & .    & true & . & 3 & true & true & true & 14 & 16 & 0.533333 & 132 & 0 & 0 & 132 &  48 & 34 & 0.415 &  82 & 5.68 & 0.000000 & 3.375 & -1.65,-0.7,-0.7,-0.7    & 5.68 & 5.462911 & 1.000 & 0.92850 & 29.4 & 0.5692 & 0.92106 & D & 0.6901 & 0.90079 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.215,0.803 & 0.029,0.012,0.032,0.016 & 0.175,0.506 & B,P & T & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 2 & c.38G>A  & p.Gly13Asp & 102 & 1119 &  38 & 570 & 13 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\t 12 & 25398284 & 25398284 & MMRF\\_1078\\_1\\_BM & missense\\_variant & KRAS & rs121913529 & C & T & . & true & . & true & true & . & 3 & true & true & true & 6  & 1  & 0.142857 & 138 & 0 & 0 & 138 &  75 &  7 & 0.085 &  82 & 5.68 & 0.000000 & 3.245 & -1.19,-1.19,-1.19,-1.19 & 5.68 & 4.690595 & 1.000 & 0.83872 & 25.8 & 0.2670 & 0.87870 & D & 0.5809 & 0.85919 & D & 10 & Small\\_GTP-binding\\_protein\\_domain\\_\\_1\\_ & 0.385,0.517 & 0.01,0.018,0.01,0.024   & 0.257,0.361 & B,B & T & MODERATE & ENSG00000133703 & transcript & ENST00000256078 & protein\\_coding & 2 & c.35G>A  & p.Gly12Asp &  99 & 1119 &  35 & 570 & 12 & 189 & 0 & . & . & . & . & .\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 72\n",
       "\n",
       "| chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | sample &lt;chr&gt; | EFFECT &lt;chr&gt; | GENE &lt;chr&gt; | ID &lt;chr&gt; | REF &lt;chr&gt; | ALT &lt;chr&gt; | X1000G &lt;chr&gt; | ⋯ ⋯ | CDS_POS &lt;int&gt; | CDS_LEN &lt;int&gt; | AA_POS &lt;int&gt; | AA_LEN &lt;int&gt; | DISTANCE &lt;int&gt; | ERRORS &lt;chr&gt; | LOF_GENE &lt;chr&gt; | LOF_GENEID &lt;chr&gt; | LOF_NUMTR &lt;chr&gt; | LOF_PERC &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 12 | 25380274 | 25380274 | MMRF_1021_1_BM | missense_variant | KRAS | .           | C | T | . | ⋯ | 184 | 570 | 62 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25380276 | 25380276 | MMRF_1021_1_BM | missense_variant | KRAS | rs121913240 | T | A | . | ⋯ | 182 | 570 | 61 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25398282 | 25398282 | MMRF_1030_1_BM | missense_variant | KRAS | rs121913535 | C | A | . | ⋯ |  37 | 570 | 13 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25380275 | 25380275 | MMRF_1032_1_BM | missense_variant | KRAS | rs17851045  | T | G | . | ⋯ | 183 | 570 | 61 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25398281 | 25398281 | MMRF_1038_1_BM | missense_variant | KRAS | rs112445441 | C | T | . | ⋯ |  38 | 570 | 13 | 189 | 0 | . | . | . | . | . |\n",
       "| 12 | 25398284 | 25398284 | MMRF_1078_1_BM | missense_variant | KRAS | rs121913529 | C | T | . | ⋯ |  35 | 570 | 12 | 189 | 0 | . | . | . | . | . |\n",
       "\n"
      ],
      "text/plain": [
       "  chr start    end      sample         EFFECT           GENE ID          REF\n",
       "1 12  25380274 25380274 MMRF_1021_1_BM missense_variant KRAS .           C  \n",
       "2 12  25380276 25380276 MMRF_1021_1_BM missense_variant KRAS rs121913240 T  \n",
       "3 12  25398282 25398282 MMRF_1030_1_BM missense_variant KRAS rs121913535 C  \n",
       "4 12  25380275 25380275 MMRF_1032_1_BM missense_variant KRAS rs17851045  T  \n",
       "5 12  25398281 25398281 MMRF_1038_1_BM missense_variant KRAS rs112445441 C  \n",
       "6 12  25398284 25398284 MMRF_1078_1_BM missense_variant KRAS rs121913529 C  \n",
       "  ALT X1000G ⋯ CDS_POS CDS_LEN AA_POS AA_LEN DISTANCE ERRORS LOF_GENE\n",
       "1 T   .      ⋯ 184     570     62     189    0        .      .       \n",
       "2 A   .      ⋯ 182     570     61     189    0        .      .       \n",
       "3 A   .      ⋯  37     570     13     189    0        .      .       \n",
       "4 G   .      ⋯ 183     570     61     189    0        .      .       \n",
       "5 T   .      ⋯  38     570     13     189    0        .      .       \n",
       "6 T   .      ⋯  35     570     12     189    0        .      .       \n",
       "  LOF_GENEID LOF_NUMTR LOF_PERC\n",
       "1 .          .         .       \n",
       "2 .          .         .       \n",
       "3 .          .         .       \n",
       "4 .          .         .       \n",
       "5 .          .         .       \n",
       "6 .          .         .       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "variant4 = filter(variant2, GENE == 'KRAS')\n",
    "head(variant4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'missense_variant'</li>\n",
       "\t<li>'3_prime_UTR_variant'</li>\n",
       "\t<li>'intron_variant'</li>\n",
       "\t<li>'missense_variant&amp;splice_region_variant'</li>\n",
       "\t<li>'upstream_gene_variant'</li>\n",
       "\t<li>'downstream_gene_variant'</li>\n",
       "\t<li>'intragenic_variant'</li>\n",
       "\t<li>'non_coding_exon_variant'</li>\n",
       "\t<li>'5_prime_UTR_premature_start_codon_gain_variant'</li>\n",
       "\t<li>'splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'synonymous_variant'</li>\n",
       "\t<li>'frameshift_variant'</li>\n",
       "\t<li>'5_prime_UTR_variant'</li>\n",
       "\t<li>'stop_gained'</li>\n",
       "\t<li>'splice_region_variant&amp;synonymous_variant'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;intron_variant'</li>\n",
       "\t<li>'splice_donor_variant&amp;intron_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_deletion'</li>\n",
       "\t<li>'start_lost'</li>\n",
       "\t<li>'splice_region_variant&amp;non_coding_exon_variant'</li>\n",
       "\t<li>'splice_region_variant'</li>\n",
       "\t<li>'stop_retained_variant'</li>\n",
       "\t<li>'stop_gained&amp;splice_region_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;stop_gained'</li>\n",
       "\t<li>'splice_donor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost'</li>\n",
       "\t<li>'inframe_insertion'</li>\n",
       "\t<li>'inframe_deletion'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_donor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'splice_donor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost&amp;splice_region_variant'</li>\n",
       "\t<li>'initiator_codon_variant'</li>\n",
       "\t<li>'disruptive_inframe_insertion'</li>\n",
       "\t<li>'inframe_deletion&amp;splice_region_variant'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost&amp;inframe_deletion&amp;splice_region_variant'</li>\n",
       "\t<li>'stop_gained&amp;disruptive_inframe_deletion'</li>\n",
       "\t<li>'splice_donor_variant&amp;inframe_deletion&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;start_lost'</li>\n",
       "\t<li>'stop_gained&amp;disruptive_inframe_insertion'</li>\n",
       "\t<li>'inframe_insertion&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_deletion&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_insertion&amp;splice_region_variant'</li>\n",
       "\t<li>'start_lost&amp;splice_region_variant'</li>\n",
       "\t<li>'start_lost&amp;inframe_deletion'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;splice_region_variant&amp;3_prime_UTR_variant&amp;intron_variant'</li>\n",
       "\t<li>'splice_donor_variant&amp;splice_region_variant&amp;intron_variant&amp;non_coding_exon_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;stop_lost'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost&amp;inframe_deletion'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'missense\\_variant'\n",
       "\\item '3\\_prime\\_UTR\\_variant'\n",
       "\\item 'intron\\_variant'\n",
       "\\item 'missense\\_variant\\&splice\\_region\\_variant'\n",
       "\\item 'upstream\\_gene\\_variant'\n",
       "\\item 'downstream\\_gene\\_variant'\n",
       "\\item 'intragenic\\_variant'\n",
       "\\item 'non\\_coding\\_exon\\_variant'\n",
       "\\item '5\\_prime\\_UTR\\_premature\\_start\\_codon\\_gain\\_variant'\n",
       "\\item 'splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'synonymous\\_variant'\n",
       "\\item 'frameshift\\_variant'\n",
       "\\item '5\\_prime\\_UTR\\_variant'\n",
       "\\item 'stop\\_gained'\n",
       "\\item 'splice\\_region\\_variant\\&synonymous\\_variant'\n",
       "\\item 'splice\\_acceptor\\_variant\\&intron\\_variant'\n",
       "\\item 'splice\\_donor\\_variant\\&intron\\_variant'\n",
       "\\item 'frameshift\\_variant\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_deletion'\n",
       "\\item 'start\\_lost'\n",
       "\\item 'splice\\_region\\_variant\\&non\\_coding\\_exon\\_variant'\n",
       "\\item 'splice\\_region\\_variant'\n",
       "\\item 'stop\\_retained\\_variant'\n",
       "\\item 'stop\\_gained\\&splice\\_region\\_variant'\n",
       "\\item 'frameshift\\_variant\\&stop\\_gained'\n",
       "\\item 'splice\\_donor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost'\n",
       "\\item 'inframe\\_insertion'\n",
       "\\item 'inframe\\_deletion'\n",
       "\\item 'frameshift\\_variant\\&splice\\_acceptor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'frameshift\\_variant\\&splice\\_donor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'splice\\_donor\\_variant\\&disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost\\&splice\\_region\\_variant'\n",
       "\\item 'initiator\\_codon\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_insertion'\n",
       "\\item 'inframe\\_deletion\\&splice\\_region\\_variant'\n",
       "\\item 'splice\\_acceptor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost\\&inframe\\_deletion\\&splice\\_region\\_variant'\n",
       "\\item 'stop\\_gained\\&disruptive\\_inframe\\_deletion'\n",
       "\\item 'splice\\_donor\\_variant\\&inframe\\_deletion\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'frameshift\\_variant\\&start\\_lost'\n",
       "\\item 'stop\\_gained\\&disruptive\\_inframe\\_insertion'\n",
       "\\item 'inframe\\_insertion\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_insertion\\&splice\\_region\\_variant'\n",
       "\\item 'start\\_lost\\&splice\\_region\\_variant'\n",
       "\\item 'start\\_lost\\&inframe\\_deletion'\n",
       "\\item 'splice\\_acceptor\\_variant\\&splice\\_region\\_variant\\&3\\_prime\\_UTR\\_variant\\&intron\\_variant'\n",
       "\\item 'splice\\_donor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant\\&non\\_coding\\_exon\\_variant'\n",
       "\\item 'frameshift\\_variant\\&stop\\_lost'\n",
       "\\item 'splice\\_acceptor\\_variant\\&disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost\\&inframe\\_deletion'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'missense_variant'\n",
       "2. '3_prime_UTR_variant'\n",
       "3. 'intron_variant'\n",
       "4. 'missense_variant&amp;splice_region_variant'\n",
       "5. 'upstream_gene_variant'\n",
       "6. 'downstream_gene_variant'\n",
       "7. 'intragenic_variant'\n",
       "8. 'non_coding_exon_variant'\n",
       "9. '5_prime_UTR_premature_start_codon_gain_variant'\n",
       "10. 'splice_region_variant&amp;intron_variant'\n",
       "11. 'synonymous_variant'\n",
       "12. 'frameshift_variant'\n",
       "13. '5_prime_UTR_variant'\n",
       "14. 'stop_gained'\n",
       "15. 'splice_region_variant&amp;synonymous_variant'\n",
       "16. 'splice_acceptor_variant&amp;intron_variant'\n",
       "17. 'splice_donor_variant&amp;intron_variant'\n",
       "18. 'frameshift_variant&amp;splice_region_variant'\n",
       "19. 'disruptive_inframe_deletion'\n",
       "20. 'start_lost'\n",
       "21. 'splice_region_variant&amp;non_coding_exon_variant'\n",
       "22. 'splice_region_variant'\n",
       "23. 'stop_retained_variant'\n",
       "24. 'stop_gained&amp;splice_region_variant'\n",
       "25. 'frameshift_variant&amp;stop_gained'\n",
       "26. 'splice_donor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "27. 'stop_lost'\n",
       "28. 'inframe_insertion'\n",
       "29. 'inframe_deletion'\n",
       "30. 'frameshift_variant&amp;splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "31. 'frameshift_variant&amp;splice_donor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "32. 'splice_donor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'\n",
       "33. 'stop_lost&amp;splice_region_variant'\n",
       "34. 'initiator_codon_variant'\n",
       "35. 'disruptive_inframe_insertion'\n",
       "36. 'inframe_deletion&amp;splice_region_variant'\n",
       "37. 'splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "38. 'stop_lost&amp;inframe_deletion&amp;splice_region_variant'\n",
       "39. 'stop_gained&amp;disruptive_inframe_deletion'\n",
       "40. 'splice_donor_variant&amp;inframe_deletion&amp;splice_region_variant&amp;intron_variant'\n",
       "41. 'frameshift_variant&amp;start_lost'\n",
       "42. 'stop_gained&amp;disruptive_inframe_insertion'\n",
       "43. 'inframe_insertion&amp;splice_region_variant'\n",
       "44. 'disruptive_inframe_deletion&amp;splice_region_variant'\n",
       "45. 'disruptive_inframe_insertion&amp;splice_region_variant'\n",
       "46. 'start_lost&amp;splice_region_variant'\n",
       "47. 'start_lost&amp;inframe_deletion'\n",
       "48. 'splice_acceptor_variant&amp;splice_region_variant&amp;3_prime_UTR_variant&amp;intron_variant'\n",
       "49. 'splice_donor_variant&amp;splice_region_variant&amp;intron_variant&amp;non_coding_exon_variant'\n",
       "50. 'frameshift_variant&amp;stop_lost'\n",
       "51. 'splice_acceptor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'\n",
       "52. 'stop_lost&amp;inframe_deletion'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"missense_variant\"                                                                        \n",
       " [2] \"3_prime_UTR_variant\"                                                                     \n",
       " [3] \"intron_variant\"                                                                          \n",
       " [4] \"missense_variant&splice_region_variant\"                                                  \n",
       " [5] \"upstream_gene_variant\"                                                                   \n",
       " [6] \"downstream_gene_variant\"                                                                 \n",
       " [7] \"intragenic_variant\"                                                                      \n",
       " [8] \"non_coding_exon_variant\"                                                                 \n",
       " [9] \"5_prime_UTR_premature_start_codon_gain_variant\"                                          \n",
       "[10] \"splice_region_variant&intron_variant\"                                                    \n",
       "[11] \"synonymous_variant\"                                                                      \n",
       "[12] \"frameshift_variant\"                                                                      \n",
       "[13] \"5_prime_UTR_variant\"                                                                     \n",
       "[14] \"stop_gained\"                                                                             \n",
       "[15] \"splice_region_variant&synonymous_variant\"                                                \n",
       "[16] \"splice_acceptor_variant&intron_variant\"                                                  \n",
       "[17] \"splice_donor_variant&intron_variant\"                                                     \n",
       "[18] \"frameshift_variant&splice_region_variant\"                                                \n",
       "[19] \"disruptive_inframe_deletion\"                                                             \n",
       "[20] \"start_lost\"                                                                              \n",
       "[21] \"splice_region_variant&non_coding_exon_variant\"                                           \n",
       "[22] \"splice_region_variant\"                                                                   \n",
       "[23] \"stop_retained_variant\"                                                                   \n",
       "[24] \"stop_gained&splice_region_variant\"                                                       \n",
       "[25] \"frameshift_variant&stop_gained\"                                                          \n",
       "[26] \"splice_donor_variant&splice_region_variant&intron_variant\"                               \n",
       "[27] \"stop_lost\"                                                                               \n",
       "[28] \"inframe_insertion\"                                                                       \n",
       "[29] \"inframe_deletion\"                                                                        \n",
       "[30] \"frameshift_variant&splice_acceptor_variant&splice_region_variant&intron_variant\"         \n",
       "[31] \"frameshift_variant&splice_donor_variant&splice_region_variant&intron_variant\"            \n",
       "[32] \"splice_donor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\"   \n",
       "[33] \"stop_lost&splice_region_variant\"                                                         \n",
       "[34] \"initiator_codon_variant\"                                                                 \n",
       "[35] \"disruptive_inframe_insertion\"                                                            \n",
       "[36] \"inframe_deletion&splice_region_variant\"                                                  \n",
       "[37] \"splice_acceptor_variant&splice_region_variant&intron_variant\"                            \n",
       "[38] \"stop_lost&inframe_deletion&splice_region_variant\"                                        \n",
       "[39] \"stop_gained&disruptive_inframe_deletion\"                                                 \n",
       "[40] \"splice_donor_variant&inframe_deletion&splice_region_variant&intron_variant\"              \n",
       "[41] \"frameshift_variant&start_lost\"                                                           \n",
       "[42] \"stop_gained&disruptive_inframe_insertion\"                                                \n",
       "[43] \"inframe_insertion&splice_region_variant\"                                                 \n",
       "[44] \"disruptive_inframe_deletion&splice_region_variant\"                                       \n",
       "[45] \"disruptive_inframe_insertion&splice_region_variant\"                                      \n",
       "[46] \"start_lost&splice_region_variant\"                                                        \n",
       "[47] \"start_lost&inframe_deletion\"                                                             \n",
       "[48] \"splice_acceptor_variant&splice_region_variant&3_prime_UTR_variant&intron_variant\"        \n",
       "[49] \"splice_donor_variant&splice_region_variant&intron_variant&non_coding_exon_variant\"       \n",
       "[50] \"frameshift_variant&stop_lost\"                                                            \n",
       "[51] \"splice_acceptor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\"\n",
       "[52] \"stop_lost&inframe_deletion\"                                                              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'missense_variant'</li>\n",
       "\t<li>'missense_variant&amp;splice_region_variant'</li>\n",
       "\t<li>'frameshift_variant'</li>\n",
       "\t<li>'stop_gained'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_deletion'</li>\n",
       "\t<li>'start_lost'</li>\n",
       "\t<li>'stop_gained&amp;splice_region_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;stop_gained'</li>\n",
       "\t<li>'stop_lost'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_donor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'splice_donor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_insertion'</li>\n",
       "\t<li>'stop_gained&amp;disruptive_inframe_deletion'</li>\n",
       "\t<li>'frameshift_variant&amp;start_lost'</li>\n",
       "\t<li>'stop_gained&amp;disruptive_inframe_insertion'</li>\n",
       "\t<li>'disruptive_inframe_deletion&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_insertion&amp;splice_region_variant'</li>\n",
       "\t<li>'start_lost&amp;splice_region_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;stop_lost'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost&amp;inframe_deletion'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'missense\\_variant'\n",
       "\\item 'missense\\_variant\\&splice\\_region\\_variant'\n",
       "\\item 'frameshift\\_variant'\n",
       "\\item 'stop\\_gained'\n",
       "\\item 'frameshift\\_variant\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_deletion'\n",
       "\\item 'start\\_lost'\n",
       "\\item 'stop\\_gained\\&splice\\_region\\_variant'\n",
       "\\item 'frameshift\\_variant\\&stop\\_gained'\n",
       "\\item 'stop\\_lost'\n",
       "\\item 'frameshift\\_variant\\&splice\\_acceptor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'frameshift\\_variant\\&splice\\_donor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'splice\\_donor\\_variant\\&disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_insertion'\n",
       "\\item 'stop\\_gained\\&disruptive\\_inframe\\_deletion'\n",
       "\\item 'frameshift\\_variant\\&start\\_lost'\n",
       "\\item 'stop\\_gained\\&disruptive\\_inframe\\_insertion'\n",
       "\\item 'disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_insertion\\&splice\\_region\\_variant'\n",
       "\\item 'start\\_lost\\&splice\\_region\\_variant'\n",
       "\\item 'frameshift\\_variant\\&stop\\_lost'\n",
       "\\item 'splice\\_acceptor\\_variant\\&disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost\\&inframe\\_deletion'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'missense_variant'\n",
       "2. 'missense_variant&amp;splice_region_variant'\n",
       "3. 'frameshift_variant'\n",
       "4. 'stop_gained'\n",
       "5. 'frameshift_variant&amp;splice_region_variant'\n",
       "6. 'disruptive_inframe_deletion'\n",
       "7. 'start_lost'\n",
       "8. 'stop_gained&amp;splice_region_variant'\n",
       "9. 'frameshift_variant&amp;stop_gained'\n",
       "10. 'stop_lost'\n",
       "11. 'frameshift_variant&amp;splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "12. 'frameshift_variant&amp;splice_donor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "13. 'splice_donor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'\n",
       "14. 'stop_lost&amp;splice_region_variant'\n",
       "15. 'disruptive_inframe_insertion'\n",
       "16. 'stop_gained&amp;disruptive_inframe_deletion'\n",
       "17. 'frameshift_variant&amp;start_lost'\n",
       "18. 'stop_gained&amp;disruptive_inframe_insertion'\n",
       "19. 'disruptive_inframe_deletion&amp;splice_region_variant'\n",
       "20. 'disruptive_inframe_insertion&amp;splice_region_variant'\n",
       "21. 'start_lost&amp;splice_region_variant'\n",
       "22. 'frameshift_variant&amp;stop_lost'\n",
       "23. 'splice_acceptor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'\n",
       "24. 'stop_lost&amp;inframe_deletion'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"missense_variant\"                                                                        \n",
       " [2] \"missense_variant&splice_region_variant\"                                                  \n",
       " [3] \"frameshift_variant\"                                                                      \n",
       " [4] \"stop_gained\"                                                                             \n",
       " [5] \"frameshift_variant&splice_region_variant\"                                                \n",
       " [6] \"disruptive_inframe_deletion\"                                                             \n",
       " [7] \"start_lost\"                                                                              \n",
       " [8] \"stop_gained&splice_region_variant\"                                                       \n",
       " [9] \"frameshift_variant&stop_gained\"                                                          \n",
       "[10] \"stop_lost\"                                                                               \n",
       "[11] \"frameshift_variant&splice_acceptor_variant&splice_region_variant&intron_variant\"         \n",
       "[12] \"frameshift_variant&splice_donor_variant&splice_region_variant&intron_variant\"            \n",
       "[13] \"splice_donor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\"   \n",
       "[14] \"stop_lost&splice_region_variant\"                                                         \n",
       "[15] \"disruptive_inframe_insertion\"                                                            \n",
       "[16] \"stop_gained&disruptive_inframe_deletion\"                                                 \n",
       "[17] \"frameshift_variant&start_lost\"                                                           \n",
       "[18] \"stop_gained&disruptive_inframe_insertion\"                                                \n",
       "[19] \"disruptive_inframe_deletion&splice_region_variant\"                                       \n",
       "[20] \"disruptive_inframe_insertion&splice_region_variant\"                                      \n",
       "[21] \"start_lost&splice_region_variant\"                                                        \n",
       "[22] \"frameshift_variant&stop_lost\"                                                            \n",
       "[23] \"splice_acceptor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\"\n",
       "[24] \"stop_lost&inframe_deletion\"                                                              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'missense_variant'</li>\n",
       "\t<li>'3_prime_UTR_variant'</li>\n",
       "\t<li>'intron_variant'</li>\n",
       "\t<li>'missense_variant&amp;splice_region_variant'</li>\n",
       "\t<li>'upstream_gene_variant'</li>\n",
       "\t<li>'downstream_gene_variant'</li>\n",
       "\t<li>'intragenic_variant'</li>\n",
       "\t<li>'non_coding_exon_variant'</li>\n",
       "\t<li>'5_prime_UTR_premature_start_codon_gain_variant'</li>\n",
       "\t<li>'splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'synonymous_variant'</li>\n",
       "\t<li>'frameshift_variant'</li>\n",
       "\t<li>'5_prime_UTR_variant'</li>\n",
       "\t<li>'stop_gained'</li>\n",
       "\t<li>'splice_region_variant&amp;synonymous_variant'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;intron_variant'</li>\n",
       "\t<li>'splice_donor_variant&amp;intron_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_deletion'</li>\n",
       "\t<li>'splice_region_variant&amp;non_coding_exon_variant'</li>\n",
       "\t<li>'start_lost'</li>\n",
       "\t<li>'stop_retained_variant'</li>\n",
       "\t<li>'splice_region_variant'</li>\n",
       "\t<li>'stop_gained&amp;splice_region_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;stop_gained'</li>\n",
       "\t<li>'stop_lost'</li>\n",
       "\t<li>'inframe_insertion'</li>\n",
       "\t<li>'inframe_deletion'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;splice_donor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'splice_donor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost&amp;splice_region_variant'</li>\n",
       "\t<li>'initiator_codon_variant'</li>\n",
       "\t<li>'disruptive_inframe_insertion'</li>\n",
       "\t<li>'splice_donor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'inframe_deletion&amp;splice_region_variant'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost&amp;inframe_deletion&amp;splice_region_variant'</li>\n",
       "\t<li>'stop_gained&amp;disruptive_inframe_deletion'</li>\n",
       "\t<li>'splice_donor_variant&amp;inframe_deletion&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;start_lost'</li>\n",
       "\t<li>'stop_gained&amp;disruptive_inframe_insertion'</li>\n",
       "\t<li>'inframe_insertion&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_deletion&amp;splice_region_variant'</li>\n",
       "\t<li>'disruptive_inframe_insertion&amp;splice_region_variant'</li>\n",
       "\t<li>'start_lost&amp;splice_region_variant'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;splice_region_variant&amp;3_prime_UTR_variant&amp;intron_variant'</li>\n",
       "\t<li>'splice_donor_variant&amp;splice_region_variant&amp;intron_variant&amp;non_coding_exon_variant'</li>\n",
       "\t<li>'frameshift_variant&amp;stop_lost'</li>\n",
       "\t<li>'splice_acceptor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'</li>\n",
       "\t<li>'stop_lost&amp;inframe_deletion'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'missense\\_variant'\n",
       "\\item '3\\_prime\\_UTR\\_variant'\n",
       "\\item 'intron\\_variant'\n",
       "\\item 'missense\\_variant\\&splice\\_region\\_variant'\n",
       "\\item 'upstream\\_gene\\_variant'\n",
       "\\item 'downstream\\_gene\\_variant'\n",
       "\\item 'intragenic\\_variant'\n",
       "\\item 'non\\_coding\\_exon\\_variant'\n",
       "\\item '5\\_prime\\_UTR\\_premature\\_start\\_codon\\_gain\\_variant'\n",
       "\\item 'splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'synonymous\\_variant'\n",
       "\\item 'frameshift\\_variant'\n",
       "\\item '5\\_prime\\_UTR\\_variant'\n",
       "\\item 'stop\\_gained'\n",
       "\\item 'splice\\_region\\_variant\\&synonymous\\_variant'\n",
       "\\item 'splice\\_acceptor\\_variant\\&intron\\_variant'\n",
       "\\item 'splice\\_donor\\_variant\\&intron\\_variant'\n",
       "\\item 'frameshift\\_variant\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_deletion'\n",
       "\\item 'splice\\_region\\_variant\\&non\\_coding\\_exon\\_variant'\n",
       "\\item 'start\\_lost'\n",
       "\\item 'stop\\_retained\\_variant'\n",
       "\\item 'splice\\_region\\_variant'\n",
       "\\item 'stop\\_gained\\&splice\\_region\\_variant'\n",
       "\\item 'frameshift\\_variant\\&stop\\_gained'\n",
       "\\item 'stop\\_lost'\n",
       "\\item 'inframe\\_insertion'\n",
       "\\item 'inframe\\_deletion'\n",
       "\\item 'frameshift\\_variant\\&splice\\_acceptor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'frameshift\\_variant\\&splice\\_donor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'splice\\_donor\\_variant\\&disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost\\&splice\\_region\\_variant'\n",
       "\\item 'initiator\\_codon\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_insertion'\n",
       "\\item 'splice\\_donor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'inframe\\_deletion\\&splice\\_region\\_variant'\n",
       "\\item 'splice\\_acceptor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost\\&inframe\\_deletion\\&splice\\_region\\_variant'\n",
       "\\item 'stop\\_gained\\&disruptive\\_inframe\\_deletion'\n",
       "\\item 'splice\\_donor\\_variant\\&inframe\\_deletion\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'frameshift\\_variant\\&start\\_lost'\n",
       "\\item 'stop\\_gained\\&disruptive\\_inframe\\_insertion'\n",
       "\\item 'inframe\\_insertion\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant'\n",
       "\\item 'disruptive\\_inframe\\_insertion\\&splice\\_region\\_variant'\n",
       "\\item 'start\\_lost\\&splice\\_region\\_variant'\n",
       "\\item 'splice\\_acceptor\\_variant\\&splice\\_region\\_variant\\&3\\_prime\\_UTR\\_variant\\&intron\\_variant'\n",
       "\\item 'splice\\_donor\\_variant\\&splice\\_region\\_variant\\&intron\\_variant\\&non\\_coding\\_exon\\_variant'\n",
       "\\item 'frameshift\\_variant\\&stop\\_lost'\n",
       "\\item 'splice\\_acceptor\\_variant\\&disruptive\\_inframe\\_deletion\\&splice\\_region\\_variant\\&intron\\_variant'\n",
       "\\item 'stop\\_lost\\&inframe\\_deletion'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'missense_variant'\n",
       "2. '3_prime_UTR_variant'\n",
       "3. 'intron_variant'\n",
       "4. 'missense_variant&amp;splice_region_variant'\n",
       "5. 'upstream_gene_variant'\n",
       "6. 'downstream_gene_variant'\n",
       "7. 'intragenic_variant'\n",
       "8. 'non_coding_exon_variant'\n",
       "9. '5_prime_UTR_premature_start_codon_gain_variant'\n",
       "10. 'splice_region_variant&amp;intron_variant'\n",
       "11. 'synonymous_variant'\n",
       "12. 'frameshift_variant'\n",
       "13. '5_prime_UTR_variant'\n",
       "14. 'stop_gained'\n",
       "15. 'splice_region_variant&amp;synonymous_variant'\n",
       "16. 'splice_acceptor_variant&amp;intron_variant'\n",
       "17. 'splice_donor_variant&amp;intron_variant'\n",
       "18. 'frameshift_variant&amp;splice_region_variant'\n",
       "19. 'disruptive_inframe_deletion'\n",
       "20. 'splice_region_variant&amp;non_coding_exon_variant'\n",
       "21. 'start_lost'\n",
       "22. 'stop_retained_variant'\n",
       "23. 'splice_region_variant'\n",
       "24. 'stop_gained&amp;splice_region_variant'\n",
       "25. 'frameshift_variant&amp;stop_gained'\n",
       "26. 'stop_lost'\n",
       "27. 'inframe_insertion'\n",
       "28. 'inframe_deletion'\n",
       "29. 'frameshift_variant&amp;splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "30. 'frameshift_variant&amp;splice_donor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "31. 'splice_donor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'\n",
       "32. 'stop_lost&amp;splice_region_variant'\n",
       "33. 'initiator_codon_variant'\n",
       "34. 'disruptive_inframe_insertion'\n",
       "35. 'splice_donor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "36. 'inframe_deletion&amp;splice_region_variant'\n",
       "37. 'splice_acceptor_variant&amp;splice_region_variant&amp;intron_variant'\n",
       "38. 'stop_lost&amp;inframe_deletion&amp;splice_region_variant'\n",
       "39. 'stop_gained&amp;disruptive_inframe_deletion'\n",
       "40. 'splice_donor_variant&amp;inframe_deletion&amp;splice_region_variant&amp;intron_variant'\n",
       "41. 'frameshift_variant&amp;start_lost'\n",
       "42. 'stop_gained&amp;disruptive_inframe_insertion'\n",
       "43. 'inframe_insertion&amp;splice_region_variant'\n",
       "44. 'disruptive_inframe_deletion&amp;splice_region_variant'\n",
       "45. 'disruptive_inframe_insertion&amp;splice_region_variant'\n",
       "46. 'start_lost&amp;splice_region_variant'\n",
       "47. 'splice_acceptor_variant&amp;splice_region_variant&amp;3_prime_UTR_variant&amp;intron_variant'\n",
       "48. 'splice_donor_variant&amp;splice_region_variant&amp;intron_variant&amp;non_coding_exon_variant'\n",
       "49. 'frameshift_variant&amp;stop_lost'\n",
       "50. 'splice_acceptor_variant&amp;disruptive_inframe_deletion&amp;splice_region_variant&amp;intron_variant'\n",
       "51. 'stop_lost&amp;inframe_deletion'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"missense_variant\"                                                                        \n",
       " [2] \"3_prime_UTR_variant\"                                                                     \n",
       " [3] \"intron_variant\"                                                                          \n",
       " [4] \"missense_variant&splice_region_variant\"                                                  \n",
       " [5] \"upstream_gene_variant\"                                                                   \n",
       " [6] \"downstream_gene_variant\"                                                                 \n",
       " [7] \"intragenic_variant\"                                                                      \n",
       " [8] \"non_coding_exon_variant\"                                                                 \n",
       " [9] \"5_prime_UTR_premature_start_codon_gain_variant\"                                          \n",
       "[10] \"splice_region_variant&intron_variant\"                                                    \n",
       "[11] \"synonymous_variant\"                                                                      \n",
       "[12] \"frameshift_variant\"                                                                      \n",
       "[13] \"5_prime_UTR_variant\"                                                                     \n",
       "[14] \"stop_gained\"                                                                             \n",
       "[15] \"splice_region_variant&synonymous_variant\"                                                \n",
       "[16] \"splice_acceptor_variant&intron_variant\"                                                  \n",
       "[17] \"splice_donor_variant&intron_variant\"                                                     \n",
       "[18] \"frameshift_variant&splice_region_variant\"                                                \n",
       "[19] \"disruptive_inframe_deletion\"                                                             \n",
       "[20] \"splice_region_variant&non_coding_exon_variant\"                                           \n",
       "[21] \"start_lost\"                                                                              \n",
       "[22] \"stop_retained_variant\"                                                                   \n",
       "[23] \"splice_region_variant\"                                                                   \n",
       "[24] \"stop_gained&splice_region_variant\"                                                       \n",
       "[25] \"frameshift_variant&stop_gained\"                                                          \n",
       "[26] \"stop_lost\"                                                                               \n",
       "[27] \"inframe_insertion\"                                                                       \n",
       "[28] \"inframe_deletion\"                                                                        \n",
       "[29] \"frameshift_variant&splice_acceptor_variant&splice_region_variant&intron_variant\"         \n",
       "[30] \"frameshift_variant&splice_donor_variant&splice_region_variant&intron_variant\"            \n",
       "[31] \"splice_donor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\"   \n",
       "[32] \"stop_lost&splice_region_variant\"                                                         \n",
       "[33] \"initiator_codon_variant\"                                                                 \n",
       "[34] \"disruptive_inframe_insertion\"                                                            \n",
       "[35] \"splice_donor_variant&splice_region_variant&intron_variant\"                               \n",
       "[36] \"inframe_deletion&splice_region_variant\"                                                  \n",
       "[37] \"splice_acceptor_variant&splice_region_variant&intron_variant\"                            \n",
       "[38] \"stop_lost&inframe_deletion&splice_region_variant\"                                        \n",
       "[39] \"stop_gained&disruptive_inframe_deletion\"                                                 \n",
       "[40] \"splice_donor_variant&inframe_deletion&splice_region_variant&intron_variant\"              \n",
       "[41] \"frameshift_variant&start_lost\"                                                           \n",
       "[42] \"stop_gained&disruptive_inframe_insertion\"                                                \n",
       "[43] \"inframe_insertion&splice_region_variant\"                                                 \n",
       "[44] \"disruptive_inframe_deletion&splice_region_variant\"                                       \n",
       "[45] \"disruptive_inframe_insertion&splice_region_variant\"                                      \n",
       "[46] \"start_lost&splice_region_variant\"                                                        \n",
       "[47] \"splice_acceptor_variant&splice_region_variant&3_prime_UTR_variant&intron_variant\"        \n",
       "[48] \"splice_donor_variant&splice_region_variant&intron_variant&non_coding_exon_variant\"       \n",
       "[49] \"frameshift_variant&stop_lost\"                                                            \n",
       "[50] \"splice_acceptor_variant&disruptive_inframe_deletion&splice_region_variant&intron_variant\"\n",
       "[51] \"stop_lost&inframe_deletion\"                                                              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(variant$EFFECT)\n",
    "unique(variant2$EFFECT)\n",
    "unique(variant3$EFFECT)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'apples'</li>\n",
       "\t<li>'bag'</li>\n",
       "\t<li>'bag'</li>\n",
       "\t<li>'milk'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'apples'\n",
       "\\item 'bag'\n",
       "\\item 'bag'\n",
       "\\item 'milk'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'apples'\n",
       "2. 'bag'\n",
       "3. 'bag'\n",
       "4. 'milk'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"apples\" \"bag\"    \"bag\"    \"milk\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "shopping_list <- c(\"apples x4\", \"bag of flour\", \"bag of sugar\", \"milk x2\")\n",
    "str_extract(shopping_list, \"[a-z]+\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'4'</li>\n",
       "\t<li>NA</li>\n",
       "\t<li>NA</li>\n",
       "\t<li>'2'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '4'\n",
       "\\item NA\n",
       "\\item NA\n",
       "\\item '2'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '4'\n",
       "2. NA\n",
       "3. NA\n",
       "4. '2'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"4\" NA  NA  \"2\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "str_extract(shopping_list, \"[0-9]+\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
